esp/src/src/ESPUtil.ts

```diff
@@ -284,14 +284,38 @@ export var LocalStorage = dojo.declare([Evented], {
     removeItem: function (key) {
         localStorage.removeItem(key);
     },
-    getItem: function (key) {
-        localStorage.getItem(key)
+    getItem: function (key, type) {
+        return type === "Number" ? Number(localStorage.getItem(key)) : localStorage.getItem(key);
     },
     clear: function () {
         localStorage.clear();
     }
 });
 
+export function goToPageUserPreference(gridName, key) {
+    var context = this;
+    var inGotoPage;
+    var initUserPref = this.LocalStorage.prototype.getItem(key, "Number");
+    if (initUserPref) {
+        gridName.set("rowsPerPage", initUserPref);
+    }
+    aspect.after(gridName, 'gotoPage', function (deferred, args) {
+        return deferred.then(function (){
```

Comment:
How can we prevent "This deferred has already been resolved" in this call?

Reply:
Is it happening for all grids are just the WUQuery widget?

Reply:
All grids

Reply:
This would have been happening before you relocated the code as well?

Reply:
@GordonSmith yes, it does just verified. Open a separate issue to deal with this?

Reply:
@miguelvazq you can open a new issue - but it may simply end up being a "fact of life", that the deferred is already resolved before we get handed it.

Reply:
@GordonSmith any further changes to this then?

Reply:
I would suggest making the "key" passed in match the filename of the file it came from, IOW:
* DFUQueryWidget
* GetDFUWorkunitsWidget
* QuerySetQueryWidget
* WUQueryWidget